### PR TITLE
Add build instructions for Debian Linux

### PR DESCRIPTION
**Pull request by jayaddison**
@blendogames @flibitijibibo Thanks for an awesome game and 64-bit Linux port :)

I was able to get `qc` to build under Debian Linux (bullseye, 64-bit) ~~with just one minor adjustment to `CMakeLists.txt` included in this pull request~~.

~~I'm not too familiar with `cmake` other than running it for build purposes, so please take that change with a grain of salt!  Hopefully the rest of the build instructions might be useful for anyone else building the game on Debian/Linux/similar systems.~~

screenshot

### CODE REVIEW

File: build.linux.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Build Instructions
+
```
Comment (flibitijibibo) [inline build.linux.md:2]: May be worth noting that the official system used to compile QC is with `flibitBuild`:

https://icculus.org/finger/flibitijibibo

File: CMakeLists.txt [hunk 2]
```diff
@@ -469,7 +469,7 @@ ADD_EXECUTABLE(qc ${QC_SRC})
 # SDL2 Dependency (Detection pulled from FAudio)
 if (DEFINED SDL2_INCLUDE_DIRS AND DEFINED SDL2_LIBRARIES)
 	message(STATUS "using pre-defined SDL2 variables SDL2_INCLUDE_DIRS and SDL2_LIBRARIES")
-	target_include_directories(qc PUBLIC "$<BUILD_INTERFACE:${SDL2_INCLUDE_DIRS}>")
+	target_include_directories(qc PUBLIC ${SDL2_INCLUDE_DIRS})
```
Comment (flibitijibibo) [inline CMakeLists.txt:472]: I'd like to know why this change was needed as well - this comes straight from FAudio which Debian packages, so as far as I know this should work. I ran into an issue with the includes myself, so there _is_ a difference needed but I'd like to know what it really is:

https://github.com/blendogames/quadrilateralcowboy/blob/master/idlib/precompiled.h#L122

**Review by flibitijibibo — COMMENTED** [review]
Other than the inlined comments this seems fine.

File: CMakeLists.txt (hunk 2 above)
Comment (jayaddison) [reply CMakeLists.txt]: I haven't been able to figure this one out, and have been able to build from a clean checkout without this change in place at all, so I've reverted it.

**Review by flibitijibibo — APPROVED** [review]

**Comment by jayaddison** [comment]
Ping @blendogames - let me know if this might be mergeable, or I can close this if it's not likely to be useful.

**Comment by jayaddison** [comment]
(closing for now; there's a way to recover the commits via GitHub pr refs if anyone's curious in future.  It might also make more sense to move the instructions here into actual scripts/code as part of a Debian package)

**Comment by jayaddison** [comment]
> (closing for now; there's a way to recover the commits via GitHub pr refs if anyone's curious in future. It might also make more sense to move the instructions here into actual scripts/code as part of a Debian package)

The relevant Debian bug for packaging of the `quadrilateralcowboy` engine (plus a few modifications) is: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1026277
